### PR TITLE
Add explanation of the zero-bug policy

### LINKED ISSUE #455: [Tracker]Implement a "Zero bug" policy
This has come up during the Spring retrospective and we could benefit from
formalizing what it means to decide if we want it, and how.

The following is my opinion, which does not have the weight of experience (on
that subject).

------------

TL;DR: If we can live with it, it's not a bug, it's an improvement (I stole that
one)

Alternative TL;DR: It's not about being bug free, but about being known bug
free.

At the start of the sprint, the product owner review all **confirmed** (aka
reproducible) bugs and decide for each if it's need to be fixed in the sprint:

- if it does, it's prioritized above all else
- if it does not, we don't consider it a bug:
    - either we won't fix it
    - or it will still be good to work on, it's a (potential) improvement 

The criteria should be a quantified impact on something we care about
(customers/users ? Crash failures ? etc).

I think the important part is to consider that a bug is something which we
should **imperatively** fix in a defined period of time (a sprint looks like a
good one).

The supposed benefits:

- easier issues prioritization. Bugs are removed from arbitration and always
  comes first  
- less workarounds. Ideally we should never need to work around bugs, because if
  we do, we should be working on fixing the bug instead.
- less "rethinking" of bugs: "Is this still valid/happening ?"

This relies heavily on a good triaging process for bugs. Especially, I think a
bug which does not have a clear reproducer should not be `triage/accepted`
(reproducing is part of the triage). This might sometimes be difficult though.

The criteria used for judging the bugs is also a question mark for me right now
(I don't think "customers are loosing money" apply to Thoth yet ?).

Three blog posts which I found insightful on the subject (with various takes):

- https://www.xolv.io/blog/articles/zero-bug-software-development/
- https://zoopla.blog/posts/2022/zero-bug-policy/
- https://medium.com/next-level-german-engineering/zero-bug-policy-experience-3121b3046326
- https://www.infoq.com/news/2021/11/zero-bug-policy/

@cgoern

/triage accepted

### PR DESCRIPTION
**Pull request by VannTen**
Closes #455

Should I add some examples for what should be `kind/bug` or not ?

Also, what how do we qualify stuff which is broken but "not problematic enough"
? `kind/cleanup` ? Hypothetical `kind/enhancement` ?

/hold

### CODE REVIEW

File: docs/TermsAndConditionsForTheScrum.md [hunk 1]
```diff
@@ -79,6 +79,19 @@ Applying any "priority/" label is only ok before the backlog refinement session,
...
 
+### Zero-bug policy
+
+In order to simplify planning, we implement a zero-bug policy. This means that
+we prioritize bugs above all other issues, **for a certain definition of bug**.
+During triage, SIGs should label sufficiently problematic issues (how they
+determine that is up to the SIGs) with `kind/bug`. These issues are prioritized
+for the next sprint.
+
+Other "bugs" (somethings does not work as it should) should be requalified. #TODO: as what
```
Comment (KPostOffice) [inline docs/TermsAndConditionsForTheScrum.md:90]: I don't think that re-qualifying is necessarily a good idea. This is basically a repackaging of bug triaging which is what the zero bug policy is trying to get rid of. If we want to create a zero bug policy, I think that a bug should be narrowly defined.

Comment (VannTen) [reply docs/TermsAndConditionsForTheScrum.md:90]: I tend to agree but reporters won't necessary have the same definition. So something reported as a bug but falling outside of that definition needs to be evaluated, and eventually re qualified if we consider it rather an improvement/feature.

File: docs/TermsAndConditionsForTheScrum.md [hunk 2]
```diff
@@ -79,6 +79,19 @@ Applying any "priority/" label is only ok before the backlog refinement session,
 important to the team, or important to support an upcoming key result. During the refinement session, the issue should
 be planned accordingly.
 
+### Zero-bug policy
+
+In order to simplify planning, we implement a zero-bug policy. This means that
```
Comment (goern) [inline docs/TermsAndConditionsForTheScrum.md:84]: What about bugs that needs instant action/attention?

**Comment by VannTen** [comment]
> What about bugs that needs instant action/attention?

By definition that won't be in planning right ?

If it's a simple fix we just drop what we're doing and do it; if it's
not we put in place a temporary workaround -> and makes the proper fix a
`kind/bug` to be solved so the workaround does remain temporary.

File: docs/TermsAndConditionsForTheScrum.md [hunk 3]
```diff
@@ -149,6 +162,8 @@ The outcome of the backlog refinement session is a set of issues that are proper
 
 ## Sprint Planning Session
 
+SIGs pulls the `kind/bug` issues into the `next` list.
```
Comment (codificat) [inline docs/TermsAndConditionsForTheScrum.md:165]: I would point to the policy here:
```suggestion
Due to the <a href="#zero-bug-policy">Zero-bug policy</a>, first SIGs pulls the (already qualified) `kind/bug` issues into the `next` list.
```

**Review by harshad16 — APPROVED** [review]
/lgtm
thanks :100:

**Pull request merged**